### PR TITLE
Fluent:  sendbox transcript navigation

### LINKED ISSUE #5153: [Fluent UI] Keyboard navigational keys not working on send box
### Is it an issue related to Adaptive Cards?

No

### Is this an accessibility issue?

No

### What version of Web Chat are you using?

Development build

### Which distribution are you using Web Chat from?

Unrelated

### Which hosting environment does this issue primarily affect?

Others or unrelated

### Which browsers and platforms do the issue happened?

Others or unrelated

### Which area does this issue affect?

Send box

### Which theme pack does this issue affect?

Fluent UI

### What is the public URL for the website?

_No response_

### Please describe the bug

When pressing navigational keys (<kbd>PGUP</kbd>, <kbd>PGDN</kbd>, <kbd>HOME</kbd>, <kbd>END</kbd>) on an empty send box, it should scroll the chat history.

### Do you see any errors in console log?

_No response_

### How to reproduce the issue?

1. Enable Fluent UI theme
1. Send "help" to MockBot
1. EXPECT: It should scroll to bottom with a few pages of chat history
1. Press <kbd>PGUP</kbd>

### What do you expect?

It should scroll up the chat history.

### What actually happened?

It did not scroll up.

### Do you have any screenshots or recordings to repro the issue?

https://github.com/microsoft/BotFramework-WebChat/assets/1622400/944fd73c-bad3-4906-9fc7-68012be12a35

### Adaptive Card JSON

_No response_

### Additional context

If the send box is not empty, navigational keys should not scroll the chat history.

### PR DESCRIPTION
**Pull request by OEvgeny**
> Fixes #5153

## Changelog Entry

- Added missing support for chat history scroll with keyboard when Fluent send box is focused, in PR [5191](https://github.com/microsoft/BotFramework-WebChat/pull/5191), by [@OEvgeny](https://github.com/OEvgeny)

## Description

This adds necessary handlers for transcript navigation into Fluent SendBox for feature parity with white label experience.

## Design

Took this as an opportunity to reduce context usage. The subscriptions are now handled via `use-paginate`.

This adds the new `useTranscriptNavigation` private hook which currently returns only the `keydown` event handler, but can be further extended in the future. The handler is designed to be placed in the send box root and is intended to handle only usage patterns seen in the send box. 

## Specific Changes

- Reworked `useScrollRelative` and `useRegisterScrollRelative`
- Added transcript scroll with keyboard support into Fluent SendBox component
- Added tests

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [x] ~I have updated documentation~

## Review Checklist

> This section is for contributors to review your work.

-  [x] ~Accessibility reviewed (tab order, content readability, alt text, color contrast)~
-  [x] ~Browser and platform compatibilities reviewed~
-  [x] ~CSS styles reviewed (minimal rules, no `z-index`)~
-  [x] ~Documents reviewed (docs, samples, live demo)~
-  [x] ~Internationalization reviewed (strings, unit formatting)~
-  [x] ~`package.json` and `package-lock.json` reviewed~
-  [x] ~Security reviewed (no data URIs, check for nonce leak)~
-  [x] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: __tests__/html/hooks.useScrollUpDown.html [hunk 1]
```diff
@@ -57,12 +57,12 @@
         expect(scrollable.scrollTop).toBe(scrollTopAtBottommost);
 
         // Call useScrollUp() should scroll the transcript view up
-        await renderWithFunction(() => useScrollUp()());
+        await renderWithFunction(() => Promise.resolve(useScrollUp()).then(fn => fn()));
```
Comment (compulim) [inline __tests__/html/hooks.useScrollUpDown.html:60]: Curious why we need to `Promise.resolve()` here.

Comment (OEvgeny) [reply __tests__/html/hooks.useScrollUpDown.html:60]: As we use `usePropagate` we can't call the callback in render. I can rework to `setImmediate` instead. I just didn't figure out how I'd use it in the context. It could be:
```
        await renderWithFunction(() => setImmediate(fn =>fn(), useScrollUp());
```
And it is [not recommended for usage](https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate)

Comment (compulim) [reply __tests__/html/hooks.useScrollUpDown.html:60]: Oh I see, you are right.

There is an escape hatch there... `allowPropagateDuringRender: true`. I think we could use it here.

tl;dr "if you promise not to re-render, I am okay that you call `usePropagate`."

Comment (compulim) [reply __tests__/html/hooks.useScrollUpDown.html:60]: My bad... we shouldn't allow propagate in almost/all circumstances.

**Review by compulim — APPROVED** [review]
